Enterprise-CMCS/macpro-mdct-mcr
# backend update to allow for YoY copy

### Description

This PR updates the create report handler to accept an optional `copySourceId` request body value.

If `copySourceId` is passed, create handler should:
- Get the field data and form template for report with id
- If the form template version is different than the newest one, compute a list of all possible field names for each form template version
  - Remove any field data keys from the field data that do not exist in the “new” form template.
  - Combine this copied over field data to the newly created field data for the report.
  - Copied over data should not replace the new field data—use object destructuring to combine.
- Finish report creation as normal.

### Related ticket(s)


MDCT-2771
---
### How to test

MCPAR - create a new program, it should work. It should not fail. 
Currently, there's not a front end update to request a report to copy, so that will come next


### Important updates




---
### Author checklist


---


_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_
